Docs: update OS settings status explanation (#25782)
fleetdm/fleet: articles/custom-os-settings.md

In the top box, with "Verified," "Verifying," "Pending," and "Failed" statuses, click each status to view a list of hosts:

> * **Verified**: hosts that applied all OS settings. Fleet verified with osquery on macOS and Windows hosts (declarations are verified with [DDM](https://support.apple.com/en-gb/guide/deployment/depb1bab77f8/web)).

Reviewer: ```suggestion
* **Verified**: hosts that applied all OS settings. Fleet verified by running an osquery query on macOS and Windows hosts (declarations profiles are verified with [DDM](https://support.apple.com/en-gb/guide/deployment/depb1bab77f8/web)).
```
- Minor tweak to call these "declaration profiles" to be consistent with other spots in docs and UI. 
- Added details on what verified with osquery means.

What does verified with DDM mean? I think we want to explain that. 

Also what about iOS/iPadOS hosts? Those show up as verified right? I think we should be explicit and explain that we move these straight to verified when they ack the MDM command (if that's right).

Author: > Also what about iOS/iPadOS hosts? Those show up as verified right? I think we should be explicit and explain that we move these straight to verified when they ack the MDM command (if that's right).

That's true. I'll add that to docs.

> * **Verified**: hosts that applied all OS settings. Fleet verified by running an osquery query on macOS and Windows hosts (declarations profiles are verified with [DDM](https://support.apple.com/en-gb/guide/deployment/depb1bab77f8/web)). Currently, iOS and iPadOS hosts are "Verified" after they acknowledge all MDM commands to apply OS settings.

Reviewer: ```suggestion
* **Verified**: hosts that applied all OS settings. Fleet verified by running an osquery query on Windows and macOS hosts (declarations profiles are verified with [DDM](https://support.apple.com/en-gb/guide/deployment/depb1bab77f8/web)). Currently, iOS and iPadOS hosts are "Verified" after they acknowledge all MDM commands to apply OS settings.
```

Reviewer: Swapping Windows and macOS so declaration profiles callout lives next to macOS.

> * **Verified**: hosts that applied all OS settings. Fleet verified by running an osquery query on Windows and macOS hosts (declarations profiles are verified with [DDM](https://support.apple.com/en-gb/guide/deployment/depb1bab77f8/web)). Currently, iOS and iPadOS hosts are "Verified" after they acknowledge all MDM commands to apply OS settings.

Reviewer: What does "verified with DDM" mean? I think we want to try to be more explicit here.

> * **Verified**: hosts that applied all OS settings. Fleet verified by running an osquery query on Windows and macOS hosts (declarations profiles are verified with [DDM](https://support.apple.com/en-gb/guide/deployment/depb1bab77f8/web)). Currently, iOS and iPadOS hosts are "Verified" after they acknowledge all MDM commands to apply OS settings.

Author: ```suggestion
* **Verified**: hosts that applied all OS settings. Fleet verified by running an osquery query on Windows and macOS hosts (declarations profiles are verified with [StatusReport](https://developer.apple.com/documentation/devicemanagement/statusreport) that hosts return to the server to confirm that OS setting is applied). Currently, iOS and iPadOS hosts are "Verified" after they acknowledge all MDM commands to apply OS settings.
```

Is this clear? Here's [flow chart](https://www.figma.com/design/t3j8CGAHR1x1YGjuFLlMst/%2314550-Add-declaration-(DDM)-profiles-for-macOS?node-id=476-11496&t=hIxWiDyeY3LzkWpS-11) that describes how declarations work. When new declaration is applied host automatically sends StatusReport to server to confirm if settings applied through declaration are applied.

Reviewer: @marko-lisica thanks! I think the StatusReport was the key. I simplified your language.

> * **Verified**: hosts that applied all OS settings. Fleet verified by running an osquery query on macOS and Windows hosts (declarations profiles are verified with [DDM](https://support.apple.com/en-gb/guide/deployment/depb1bab77f8/web)). Currently,iOS and iPadOS hosts will have "Verified" status after they acknowledge all MDM commands to apply OS settings.

Reviewer: Small language tweak.

> * Verifying: hosts that have acknowledged all MDM commands to apply OS settings. Fleet is verifying the OS settings are applied with osquery on macOS (declarations are verified with [DDM](https://support.apple.com/en-gb/guide/deployment/depb1bab77f8/web)) and Windows hosts. If the profile wasn't installed, Fleet will redeliver the profile.

Reviewer: ```suggestion
* Verifying: hosts that have acknowledged all MDM commands to apply OS settings. Fleet is verifying the OS settings are applied with osquery on Windows and macOS hosts (declarations are verified with [DDM](https://support.apple.com/en-gb/guide/deployment/depb1bab77f8/web)). If the profile wasn't installed, Fleet will redeliver the profile.
```

> * Pending: hosts that will receive MDM commands to apply OS settings when the hosts come online.

Reviewer: ```suggestion
* Pending: hosts that will receive MDM commands to apply OS settings when the hosts come online.
```

@marko-lisica we added new language for "when the hosts come online" in the UI right? 

Something like "when the hosts come online or, if it's online, when it next checks in." 

IIRC it's not exactly that^

Author: We added [this change](https://github.com/fleetdm/fleet/blob/eb9b1d615c8880024a98e671ec4d1585d6479c0e/cmd/fleetctl/scripts.go#L167):

`Script is running or will run when the host comes online...`

I think in this case host is processing command or will receive MDM command when it comes online.

Author: ```suggestion
* Pending: hosts that are processing MDM command or will receive MDM commands to apply OS settings when they come online.
```

Author: @noahtalerman Wdyt?

Reviewer: I think let's go with "running" instead of "processing"
